### PR TITLE
Opentracing doc update

### PR DESCRIPTION
**Pull request by JorikSchellekens**
Removed a reference to old code.

Added a description for how tags are used.

### CODE REVIEW

File: synapse/logging/opentracing.py [hunk 1]
```diff
@@ -68,52 +71,64 @@
...
+Setting Tags
+------------
+
+To set a tag on the active span do
+
+.. code-block:: python
+
+   from synapse.logging.opentracing import set_tag
+
+   opentracing.set_tag(tag_name, tag_value)
```
Comment (erikjohnston) [inline synapse/logging/opentracing.py]: Either `from synapse.logging import opentracing` or `set_tag(tag_name, tag_value)` :)

Comment (erikjohnston) [reply synapse/logging/opentracing.py]: Actually, I'd prefer if we used `opentracing.set_tag` etc everywhere, as then its obvious what's going on.

Comment (JorikSchellekens) [reply synapse/logging/opentracing.py]: Good idea

Comment (erikjohnston) [reply synapse/logging/opentracing.py]: Sorry, I think we should keep it as importing `set_tag`, as rich points out its syle guide. Sorry.

File: changelog.d/5776.misc [hunk 2]
```diff
@@ -0,0 +1 @@
+Remove references to trace_deferred in opetnracing docs.
```
Comment (richvdh) [inline changelog.d/5776.misc]: opetnracing

(also, this doesn't seem a good description of the change?)

**Pull request merged**